Make mouse more responsive

ad-chaos opened the pull request:
<!-- Please note that we accept pull requests from anyone, but that does not mean it will be merged. -->

## What kind of change does this PR introduce?
- Fix (I think?)

Previously every *tick* of the mouse wheel didn't result in a scroll, now it does.
## Did this PR introduce a breaking change? 

- No (at least not intentionally)

MultisampledNight commented on src/window/mouse_manager.rs:
```diff
@@ -263,16 +256,12 @@ impl MouseManager {
             }
             .into();
-            for _ in 0..(new_y - previous_y).abs() {
+            for _ in 0..(y as i32).abs()+1 {
```
And I think the culprit for that is here. Consider the case of a very small movement under `1.0`, which is almost every scroll movement on my system. Previously, those changes accumulated in `self.scroll_position` until they were large enough to cause a difference of `1.0`. But now, every single one of these is reflected and basically `1.0` (because `0.whatever as i32` is `0` and `+ 1`), which means every single frame I'm scrolling has an effect.

I'm not too sure how to correctly handle this though. Maybe we could restore the previous behavior and multiply `y` by a configurable value to accommodate every possible system, since it probably worked fine on yours.

ad-chaos replied:
Most mouse wheels are *ticked* meaning they are not smooth, so I think that it is fine to call the `nvim_input_mouse()` api function, every tick for mouse input. I think we should have different handling of touch, touchpad and mouse instead.

For touch and touchpad, I think what we could do is to calculate `pixels per line` and when those many pixels are scrolled, we can sent a single `nvim_input_mouse()` api call. But, this is not without caveats!
Under `:h 'mousescroll'` you'll see that a single api call results in different amount of lines scrolled.
The default being `ver:3,hor:6` meaning it'll scroll 3 vertical lines or 6 horizontal lines depending on the parameters passed.

Possible solutions:
1) We scroll by 3 lines every `pixels per line` scroll
2) We scroll by 3 lines every 3*`pixels per line` scroll
3) Enforce a default of `mousescroll=ver:1,hor:1` on neovide users

MultisampledNight replied:
(sorry for the delay, lost track on this one) To be honest, I think option 3 sounds like the best one, even if I usually advocate for not touching user-configurable options at all if possible. I can't figure out a case where scrolling 3 lines at once is more desirable than a flexibly configurable sensitivity config option.

MultisampledNight reviewed (commented):
Thank you!

Fix-wise, this PR makes the view scroll _insanely_ fast at every small scroll, both on a touchscreen and with a touchpad. What's a slow 25px movement with the finger on a touchscreen results into a 120 line change.

Format-wise, consider running `cargo fmt` before committing or configure your editor of choice for doing so (tip: check `:h rustfmt-autosave`).

ad-chaos commented:
> (tip: check :h rustfmt-autosave).

Wow! Never knew the rust filetype already had such features, nice!

ad-chaos commented:
I think this should be it:
1) Added a default mouse scroll of one line (vertical) and one character (horizontal) (This is configurable)
2) Changed pixel scrolling to match what  I said [here](https://github.com/neovide/neovide/pull/1511#discussion_r957536694)

ad-chaos commented:
#2057
